.devcontainer/devcontainer.json

```diff
@@ -1,63 +1,66 @@
 {
-    "name": "ROS 2 Development Container",
-    "privileged": true,
-    "remoteUser": "pi",
-    "build": {
-      "dockerfile": "Dockerfile",
-      "args": {
-        "USERNAME": "pi"
-      }
-    },
-    "workspaceFolder": "/home/pi/geicar",
-    "workspaceMount": "source=${localWorkspaceFolder},target=/home/pi/geicar/,type=bind",
-    "customizations": {
-      "vscode": {
-        "extensions":[
-            "ms-vscode.cpptools",
-            "ms-vscode.cpptools-themes",
-            "twxs.cmake",
-            "donjayamanne.python-extension-pack",
-            "eamodio.gitlens",
-            "ms-iot.vscode-ros",
-            "ms-python.python",
-            "smilerobotics.urdf",
-            "ms-python.black-formatter",
-            "ms-python.flake8"
+  "name": "ROS 2 Development Container",
+  "privileged": true,
+  "remoteUser": "pi",
+  "build": {
+    "dockerfile": "Dockerfile",
+    "args": {
+      "USERNAME": "pi"
+    }
+  },
+  "workspaceFolder": "/home/pi/geicar",
+  "workspaceMount": "source=${localWorkspaceFolder},target=/home/pi/geicar/,type=bind",
+  "customizations": {
+    "vscode": {
+      "extensions":[
+          "ms-vscode.cpptools",
+          "ms-vscode.cpptools-themes",
+          "twxs.cmake",
+          "donjayamanne.python-extension-pack",
+          "eamodio.gitlens",
+          "ms-iot.vscode-ros",
+          "ms-python.python",
+          "smilerobotics.urdf",
+          "ms-python.black-formatter",
+          "ms-python.flake8",
+          "ms-toolsai.jupyter",
+          "ms-python.vscode-pylance",
+          "ms-toolsai.vscode-ai",
+          "pytorch.pytorch-snippets",
+          "tensorflow.vscode-tensorflow"
+      ],
+      "settings": {
+        "flake8.args": [
+          "--ignore=E501"
         ],
-        "settings": {
-          "flake8.args": [
-            "--ignore=E501"
-          ],
-          "terminal.integrated.profiles.linux": {
-              "bash (Host)": {
-                  "path": "bash"
-              }
-          },
-          "terminal.integrated.defaultProfile.linux": "bash"
-        }
+        "terminal.integrated.profiles.linux": {
+            "bash (Host)": {
+                "path": "bash"
+            }
+        },
+        "terminal.integrated.defaultProfile.linux": "bash"
       }
-    },
-    "containerEnv": {
-      "DISPLAY": "unix:0",
-      "LIBGL_ALWAYS_SOFTWARE": "1", // Needed for software rendering of opengl
-      "ROS_AUTOMATIC_DISCOVERY_RANGE": "LOCALHOST",
-      "ROS_DOMAIN_ID": "1"
-    },
-    "runArgs": [
-      "-p3000:3000",
-      "-p2222:22",
-      "-p9090:9090",
-      "-e", "DISPLAY=${env:DISPLAY}",
-      "--network=host",
-      "--name=geicar",
-      "-v=/dev:/dev",
-      // "--runtime=nvidia", // NVIDIA GPU support
```

Comment:
Did you remove this line for a reason?
It was commented so that users can choose between Intel and NVIDIA GPU support